### PR TITLE
Angular core module

### PR DESCRIPTION
**Pull request by craigburke**

### CODE REVIEW

**Comment by Schlogen** [comment]
Give me time to review before merging

**Comment by craigburke** [comment]
@Schlogen sounds good. Let me know what you think.

File: profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/app.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//= wrapped
```
Comment (Schlogen) [inline profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/app.js]: Should this have the require for /angular/angular ?

File: profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/app.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+//= require /@grails.codegen.defaultPackage.path@/core/core
+//= require /@grails.codegen.defaultPackage.path@/home/home
+//= require_self
+//= require_tree services
```
Comment (Schlogen) [inline profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/app.js]: Given the main app typically wont have angular controllers, etc directly; Does it make sense to remove the requires for controllers, directives, etc ?

File: profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/app.js (hunk 1 above)
Comment (craigburke) [inline profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/app.js]: I usually put the require for angular in core module with all the other dependencies but I don't have a strong opinion on that either way. Let me know if you want me to move that to `app.js`

**Comment by craigburke** [comment]
After some very thoughtful feedback from @Schlogen I made the following changes:

* Renamed the home module to index
* Added optional arguments for DomainServiceFactory so that arguments to $resource can be overridden for individual services.
*  Check for DomainServiceFactory when `create-ng-domain` is called and fall back to using $resource directly if it's not found

File: profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/app.js (hunk 2 above)
Comment (craigburke) [inline profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/app.js]: Missed this comment the first time. Yeah, absolutely.

File: profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/core/services/DomainServiceFactory.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+angular
+    .module("@grails.codegen.defaultPackage@.core")
+    .factory("DomainServiceFactory", DomainServiceFactory);
+
+function DomainServiceFactory($resource) {
+    return function(url, paramDefaults, actions, options) {
+        return $resource(
+            url,
+            paramDefaults || null,
+            actions || {"update": {method: "PUT"}, "list": {method: "GET", isArray: true}},
```
Comment (Schlogen) [inline profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/core/services/DomainServiceFactory.js]: For this param I would use extends. That way if the user only wants to add a method, they can keep the defaults.

Comment (craigburke) [inline profiles/angular/skeleton/grails-app/assets/javascripts/@grails.codegen.defaultPackage.path@/core/services/DomainServiceFactory.js]: That makes sense.

**Pull request merged**